### PR TITLE
[MODORDERS-1214]. Replace mod-configuration usage to use Settings API from orders-storage

### PR DESCRIPTION
**Pull request by BKadirkhodjaev**
### **Purpose**

- <https://folio-org.atlassian.net/browse/MODORDERS-1214>

### **Approach**

- TODO

### CODE REVIEW

**Comment by NikitaSedyx** [comment]
# Code Review for MODORDERS-1214: Replace mod-configuration usage with Settings API from orders-storage

## Overview

This feature replaces the usage of mod-configuration API with the new Settings API from orders-storage module. The changes involve updating permissions in module descriptors, refactoring service classes to use the new settings cache and retriever, and updating tests accordingly. The migration affects configuration loading, common settings caching, and various order-related functionality.

Files modified include:
- Module descriptor template with permission updates
- Core service classes for settings retrieval and caching
- Helper utilities and order-related services
- Comprehensive test suite updates
- New test data files for orders-storage-settings

## Suggestions

## 🔧 **Code Quality: Inconsistent Method Naming Pattern**
* **Priority**: ⚠️ High
* **File**: `src/main/java/org/folio/service/settings/CommonSettingsRetriever.java`
* **Details**: The method `loadConfiguration` was renamed to `loadConfigurations` (plural), but the parameter and logic suggest it loads a single configuration object. This creates confusion about the method's actual behavior.
* **Suggested Change**: Either rename to `loadConfiguration` (singular) to match the return type `JsonObject`, or update the method to actually load multiple configurations and return a collection type.

```java
// Current (confusing):
public Future<JsonObject> loadConfigurations(RequestEntry requestEntry, RequestContext requestContext)

// Suggested (clear):
public Future<JsonObject> loadConfiguration(RequestEntry requestEntry, RequestContext requestContext)
```

## 🔧 **Critical: Inconsistent Method Signatures Between Cache and Retriever**
* **Priority**: 🔥 Critical
* **File**: `src/main/java/org/folio/service/caches/CommonSettingsCache.java`
* **Details**: The cache interface method is `loadConfigurations` but the retriever has `loadConfiguration`. This creates API inconsistency and potential runtime errors.
* **Suggested Change**: Ensure both classes use the same method name - preferably the singular form `loadConfiguration` for clarity.

## 🔧 **Performance: Potential N+1 Query Pattern in Test Mocking**
* **Priority**: ⚠️ High  
* **File**: `src/test/java/org/folio/service/pieces/PiecesClaimingServiceTest.java`
* **Details**: Multiple test methods mock `commonSettingsCache.loadConfigurations()` individually for each test, but the real implementation might be called multiple times within a single operation, potentially causing performance issues.
* **Suggested Change**: Consider implementing batch loading or ensure proper caching mechanism to avoid repeated API calls.

## ♻️ **Refactoring: Replace Deprecated List Access Pattern**
* **Priority**: 🟡 Medium
* **File**: `src/test/java/org/folio/service/pieces/PiecesClaimingServiceTest.java`
* **Details**: The code has been modernized to use `getFirst()` instead of `get(0)` for accessing the first element of lists, which is good for readability and follows modern Java practices.
* **Example**: 
```java
// Old pattern:
var error = httpException.getErrors().getErrors().get(0);
var parameter = error.getParameters().get(0);

// New pattern (already implemented):
var error = httpException.getErrors().getErrors().getFirst();
var parameter = error.getParameters().getFirst();
```
* **Suggested Action**: Ensure this pattern is consistently applied throughout the codebase.

## 📝 **Permission Updates Are Correctly Aligned**
* **Priority**: 👍 Positive
* **File**: `descriptors/ModuleDescriptor-template.json`
* **Details**: The permission changes from `configuration.entries.collection.get` to `orders-storage.settings.collection.get` are consistent throughout the module descriptor and properly reflect the migration from mod-configuration to orders-storage settings API.

## 🔧 **Test Coverage: Missing Import Statement**
* **Priority**: ⚠️ High
* **File**: `src/test/java/org/folio/service/settings/CommonSettingsRetrieverTest.java`
* **Details**: The test file adds `import org.folio.service.caches.CommonSettingsCache;` but doesn't remove the reference to the now-removed constant. This suggests a test update is incomplete.
* **Suggested Change**: Verify that all references to `CommonSettingsRetriever.TENANT_LOCALE_SETTINGS` have been properly updated to `CommonSettingsCache.TENANT_LOCALE_SETTINGS`.

## ⛏️ **Test Data Organization**
* **Priority**: 🟢 Low
* **File**: `src/test/resources/mockdata/orders-storage-settings/`
* **Details**: New test data files are well-structured with realistic JSON configurations for different scenarios (empty config, invalid config, various limits).
* **Observation**: The test data covers edge cases like invalid values ("one" instead of numeric) which is good for robust testing.

## 💭 **Architecture Concern: Resource Path Changes**
* **Priority**: 🟡 Medium
* **File**: Multiple files using `ResourcePathResolver`
* **Details**: The migration from configuration entries to settings entries suggests a fundamental change in how configuration is handled. Ensure that the new settings API has the same performance characteristics and caching behavior as the previous configuration API.
* **Suggested Action**: Document the differences in behavior between the old and new APIs, especially regarding caching and error handling.

## 🔧 **Error Handling: Consistent Exception Processing**
* **Priority**: ⚠️ High
* **File**: `src/test/java/org/folio/service/pieces/PiecesClaimingServiceTest.java`
* **Details**: Test methods consistently verify error handling scenarios, but some tests have copy-paste errors in assertions where multiple pieces check the same status instead of their individual statuses.
* **Example of Issue**:
```java
// In successWithMultipleOrganizationsAndOneIntegrationDetail test:
assertEquals(ClaimingPieceResult.Status.SUCCESS, copiedSortedResults.get(1).getStatus()); // Correct
assertEquals(ClaimingPieceResult.Status.SUCCESS, copiedSortedResults.get(1).getStatus()); // Bug: should be get(2)
assertEquals(ClaimingPieceResult.Status.SUCCESS, copiedSortedResults.get(1).getStatus()); // Bug: should be get(3)
assertEquals(ClaimingPieceResult.Status.SUCCESS, copiedSortedResults.get(1).getStatus()); // Bug: should be get(4)
```
* **Suggested Change**: Fix the assertions to check the correct array indices.

## 🌱 **Future Consideration: Configuration Migration Strategy**
* **Priority**: 🟢 Low
* **Details**: This migration represents a significant change in configuration management. Consider documenting a migration guide for installations and ensuring backward compatibility or clear upgrade paths.
* **Suggested Action**: Ensure deployment scripts and documentation are updated to reflect the new permission requirements and API endpoints.

## 🔧 **Method Naming Consistency**
* **Priority**: 🟡 Medium
* **File**: `src/test/java/org/folio/service/settings/CommonSettingsRetrieverTest.java`
* **Details**: Test method names were updated to reflect the new method names but create some inconsistency:
* `shouldLoadConfigurationSuccessfully` → `shouldLoadConfigurationsSuccessfully`
* `shouldFailToLoadConfigurationWhenRestClientFails` → `shouldFailToLoadConfigurationsWhenRestClientFails`
* **Suggested Change**: Align test method names with the final decision on whether the method should be singular or plural.

# Summary

The code changes represent a well-structured migration from mod-configuration to orders-storage settings API. The permission updates are comprehensive and consistent. However, there are several areas that need attention:

**Critical Issues:**
- Method naming inconsistency between cache and retriever interfaces
- Test assertion bugs in array index validation

**Significant Improvements:**
- Modernized List access patterns using `getFirst()`
- Comprehensive test coverage for edge cases
- Well-structured test data files

**Recommendations:**
1. Fix the method naming inconsistency between `CommonSettingsCache` and `CommonSettingsRetriever`
2. Correct the test assertion bugs in `PiecesClaimingServiceTest`
3. Ensure consistent error handling and caching behavior across the migration
4. Document the migration strategy for production deployments

The overall code quality is good, with modern Java practices being adopted and comprehensive test coverage maintained throughout the refactoring.

File: src/main/java/org/folio/orders/utils/ResourcePathResolver.java [hunk 1]
```diff
@@ -102,6 +103,7 @@ public class ResourcePathResolver {
     apis.put(EXPENSE_CLASSES_URL, "/finance/expense-classes");
     apis.put(CONFIGURATION_ENTRIES, "/configurations/entries");
     apis.put(SETTINGS_ENTRIES, "/settings/entries");
+    apis.put(ORDERS_STORAGE_SETTINGS, "/orders-storage/settings");
```
Comment (Saba-Zedginidze-EPAM) [inline src/main/java/org/folio/orders/utils/ResourcePathResolver.java:106]: We already have the internal storage endpoint defined with its service implemented
